routing/index_graph.hpp

```diff
@@ -77,16 +187,169 @@ class IndexGraph final
   {
     uint32_t const jointsSize = ReadPrimitiveFromSource<uint32_t>(src);
     m_roadIndex.Deserialize(src);
-    m_jointIndex.Build(m_roadIndex, jointsSize);
+    Build(jointsSize);
+  }
+
+  Joint::Id GetJointIdForTesting(RoadPoint const & rp) const
+  {
+    return m_roadIndex.GetJointId(rp);
+  }
+
+  /// \brief Disable an edges between |from| and |to| along |featureId|.
+  /// \note Despite the fact that |from| and |to| could be connected with several edges
+  /// it's a rare case. In most cases |from| and |to| are connected with only one edge
+  /// if they are adjacent.
+  /// \note The method doesn't affect routing if |from| and |to| are not adjacent or
+  /// if one of them is equal to Joint::kInvalidId.
+  void DisableEdge(DirectedEdge const & edge)
+  {
+    m_blockedEdges.insert(edge);
+  }
+
+  void DisableAllEdges(Joint::Id from, Joint::Id to)
+  {
+    vector<pair<RoadPoint, RoadPoint>> result;
+    m_jointIndex.FindPointsWithCommonFeature(from, to, result);
+    for (auto const & p : result)
+      DisableEdge(DirectedEdge(from, to, p.first.GetFeatureId()));
+  }
+
+  /// \brief Adding a fake oneway feature with a loose end starting from joint |from|.
+  /// Geometry for the feature points is taken from |geometrySource|.
+  /// If |geometrySource| contains more than two points the feature is created
+  /// with intermediate (not joint) point(s).
+  /// \returns feature id which was added.
+  uint32_t AddFakeLooseEndFeature(Joint::Id from, vector<RoadPoint> const & geometrySource,
+                                  double speed);
+
+  /// \brief Connects joint |from| and |to| with a fake oneway feature. Geometry for the feature
+  /// points is taken from |geometrySource|. If |geometrySource| contains more than
+  /// two points the feature is created with intermediate (not joint) point(s).
+  /// \returns feature id which was added.
+  uint32_t AddFakeFeature(Joint::Id from, Joint::Id to, vector<RoadPoint> const & geometrySource,
+                          double speed);
+
+  void ApplyRestrictionNo(RestrictionInfo const & restrictionInfo);
```

Comment:
A lot of methods in IndexGraph including(ApplyRestrictionNo/Only) are public only for test
purposes. I guess it's worth moving them to private section and declare tests (test class) as friends of IndexGraph.

ApplyRestrictionNo and ApplyRestrictionOnly are big enough to be gathered together. Because of it I implemented them as two different methods. They are used (indirectly) in ApplyRestrictions().

If it comes to restrictions the only method IndexGraph needs to have public is ApplyRestrictions(). It uses Restriction::Type enum inside to process two kinds of restrictions.